autopush/router/gcmclient.py

```diff
@@ -30,14 +33,12 @@ def __init__(self, message, response):
         self.message = message
         self.retry_message = None
 
-        self.retry_after = response.headers.get('Retry-After', None)
+        self.retry_after = (
+            response.headers.getRawHeaders('Retry-After') or [None])[0]
 
-        if response.status_code != 200:
-            self.retry_message = message
-        else:
-            self._parse_response(message, response.content)
-
-    def _parse_response(self, message, content):
+    def parse_response(self, content, code, message):
+        if code in (400, 401, 404):
```

Comment:
I think the 401 case should be removed from here just to help clarify as it would have already been handled in process()